pilot/pkg/xds/ads.go

```diff
@@ -578,11 +579,23 @@ func (s *DiscoveryServer) preProcessRequest(proxy *model.Proxy, req *discovery.D
 				event.Healthy = false
 				event.Message = req.ErrorDetail.Message
 			}
-			s.WorkloadEntryController.UpdateWorkloadEntryHealth(proxy, event)
+			err := s.WorkloadEntryController.UpdateWorkloadEntryHealth(con.proxy, event)
+			if err != nil {
```

Comment:
I am not sure we need a NACK. The point of the NACK is to tell the agent to re-send the config. But we are already returning an error immediately after, which will make the agent disconnect and re-send it on reconnection (possibly to a different instance).

So I think we want NACK *or* sending an error. Sending an error seems easier since it automatically handles backoff (I think in your case we may send NACKs as fast as possible?), but it does close the whole stream.

Reply:
In this pr, it does not close the stream when update healthcheck info failed. 
>Sending an error seems easier since it automatically handles backoff 
In xds proxy we do not have backoff policy, unless envoy does. As you can see, if we close the stream, the cost is big, all the chain would be re-established.  

I would prefer not coupling business error with network connections.

Reply:
by returning error it closes the stream? And this is returning error now

Reply:
are you reading L592?

Reply:
It will close stream when sending the response error

Reply:
Ah, understood